text/0000-contingency-plan.md

```diff
@@ -0,0 +1,425 @@
+than they were doing before. In cases like these, the same principle
+of using mitigation measures to lessen the impact (and ease the
+transition) applies, but the precise strategy to be used will have to
+be worked out on a more case-by-case basis. This is particularly
+relevant to the underspecified areas of the language described in the
+next section.
+
+Our approach to handling [dynamic drop][RFC 320] is a good
+example. Because we expect that moving to the complete non-zeroing
+dynamic drop semantics will break code, we've made an intermediate
+change that
+[altered the compiler to fill with use a non-zero value](https://github.com/rust-lang/rust/pull/23535),
+which helps to expose code that was implicitly relying on the current
+behavior (much of which has since been restructured in a more
+future-proof way).
+
+#### Underspecified language semantics
+
+There are a number of areas where the precise language semantics are
+currently somewhat underspecified. Over time, we expect to be fully
+defining the semantics of all of these areas. This may cause some
+existing code -- and in particular existing unsafe code -- to break or
+become invalid. Changes of this nature should be treated as soundness
+changes, meaning that we should attempt to mitigate the impact and
+ease the transition wherever possible.
+
+Known areas where change is expected include the following:
+
+- Destructors semantics:
+  - We plan to stop zeroing data and instead use marker flags on the stack,
+    as specified in [RFC 320]. This may affect destructors that rely on ovewriting
+    memory or using the `unsafe_no_drop_flag` attribute.
+  - Currently, panicing in a destructor can cause unintentional memory
+    leaks and other poor behavior (see [#14875], [#16135]). We are
+    likely to make panic in a destructor simply abort, but the precise
+    mechanism is not yet decided.
+  - Order of dtor execution within a data structure is somewhat
+    inconsistent (see [#744]).
+- The legal aliasing rules between unsafe pointers is not fully settled (see [#19733]).
+- The interplay of assoc types and lifetimes is not fully settled and can lead
+  to unsoundness in some cases (see [#23442]).
+- The trait selection algorithm is expected to be improved and made more complete over time.
+  It is possible that this will affect existing code.
+- [Overflow semantics][RFC 560]: in particular, we may have missed some cases.
+- Memory allocation in unsafe code is currently unstable. We expect to
+  be defining safe interfaces as part of the work on supporting
+  tracing garbage collectors (see [#415]).
+- The treatment of hygiene in macros is uneven (see [#22462], [#24278]). In some cases,
+  changes here may be backwards compatible, or may be more appropriate only with explicit opt-in
+  (or perhaps an alternate macro system altogether).
+- The layout of data structures is expected to change over time unless they are annotated
```

Comment:
This does not feel underspecifed to me. The specification is precise and complete - if you're not using `#[repr(C)]` the compiler can do what it likes with data layout and you cannot rely on it.